src/daft-sql/src/modules/uri/url_download.rs

```diff
@@ -0,0 +1,58 @@
+use daft_dsl::ExprRef;
+use daft_functions::uri::{self, download::UrlDownloadArgs};
+use sqlparser::ast::FunctionArg;
+
+use crate::{
+    error::{PlannerError, SQLPlannerResult},
+    functions::{self, SQLFunction, SQLFunctionArguments},
+    unsupported_sql_err, SQLPlanner,
+};
+
+/// The Daft-SQL `url_download` definition.
+pub struct SqlUrlDownload;
+
+impl TryFrom<SQLFunctionArguments> for UrlDownloadArgs {
+    type Error = PlannerError;
+
+    fn try_from(args: SQLFunctionArguments) -> Result<Self, Self::Error> {
+        let max_connections: usize = args.try_get_named("max_connections")?.unwrap_or(32);
+        let raise_error_on_failure = functions::args::parse_on_error(&args)?;
+        let io_config = functions::args::parse_io_config(&args)?;
+        Ok(Self {
+            max_connections,
+            raise_error_on_failure,
+            multi_thread: true, // TODO always true
```

Comment:
I wasn't able to find other examples of multi_thread in daft-sql, but the python logic defined in ExpressionUrlNamespace has

```python
multi_thread = not using_ray_runner
```